chatbot-core/backend/app/repositories/connector.py

```diff
@@ -0,0 +1,105 @@
+from sqlalchemy.orm import Session
+from typing import List, Tuple
+
+from app.models.api import APIError
+from app.models.connector import Connector
+from app.utils.error_handler import ErrorCodesMappingNumber
+from app.utils.logger import LoggerFactory
+
+logger = LoggerFactory().get_logger(__name__)
+
+
+class ConnectorRepository:
+    def __init__(self, db_session: Session):
+        self._db_session = db_session
+
+    def get_connectors(self) -> Tuple[List[Connector], APIError | None]:
+        """
+        Get all connectors
+
+        Returns:
+            Tuple[List[Connector], APIError | None]: List of connector objects and APIError object if any error
+        """
+        try:
+            connectors = self._db_session.query(Connector).all()
+            return connectors, None
+        except Exception as e:
+            logger.error(f"Error getting connectors: {e}")
+            return [], APIError(kind=ErrorCodesMappingNumber.INTERNAL_SERVER_ERROR.value)
+
+    def get_connector(self, connector_id: int) -> Tuple[Connector, APIError | None]:
+        """
+        Get connector by connector_id
+
+        Args:
+            connector_id(int): Connector id
+
+        Returns:
+            Tuple[Connector, APIError | None]: Connector object and APIError object if any error
+        """
+        try:
+            connector = self._db_session.query(Connector).filter(Connector.id == connector_id).first()
+            if not connector:
+                return None, APIError(kind=ErrorCodesMappingNumber.NOT_FOUND.value)
+            return connector, None
+        except Exception as e:
+            logger.error(f"Error getting connector: {e}")
+            return None, APIError(kind=ErrorCodesMappingNumber.INTERNAL_SERVER_ERROR.value)
+
+    def create_connector(self, connector: Connector) -> APIError | None:
```

Comment:
Improve by add 1 more function to manage transaction context.
```python
    @contextmanager
    def _transaction(self):
        """Context manager for transaction handling"""
        try:
            yield
            self._db_session.commit()
        except Exception:
            self._db_session.rollback()
            raise

    def create_connector(self, connector: Connector) -> Optional[APIError]:
        """
        Create connector with transaction management
        
        Args:
            connector: Valid Connector instance
            
        Returns:
            Optional[APIError]: None if successful, error otherwise
        """
        if not isinstance(connector, Connector):
            return APIError(kind=ErrorCodesMappingNumber.BAD_REQUEST.value)
            
        try:
            with self._transaction():
                self._db_session.add(connector)
            return None
        except SQLAlchemyError as e:
            logger.error(f"Database error creating connector: {e}")
            return APIError(kind=ErrorCodesMappingNumber.INTERNAL_SERVER_ERROR.value)
```

Reply:
Implemented. In testing progress ...